src/utils/SharedUtils.js

```diff
@@ -0,0 +1,163 @@
+export function serialize(obj, prefix) {
+  var str = [];
+  for(var p in obj) {
+    if (obj.hasOwnProperty(p)) {
+      var k = prefix ? prefix + "[" + p + "]" : p, v = obj[p];
+      str.push(typeof v == "object" ?
+        this.serialize(v, k) :
+        encodeURIComponent(k) + "=" + encodeURIComponent(v));
+    }
+  }
+  return str.join("&");
+}
+
+export function ajaxGet(url, type, success, error) {
+  var xhr = new XMLHttpRequest();
+  xhr.open('GET', url, true);
+  xhr.responseType = type;
+  xhr.setRequestHeader("Content-Type", "application/json");
+  xhr.setRequestHeader("Accept", "application/json");
+  xhr.setRequestHeader("X-Requested-With", "XMLHttpRequest");
+  xhr.onload = function() {
+    var status = xhr.status;
+    var response = xhr.response;
+    if (status === 200) {
+      return success(response);
+    } else {
+      return error(response);
+    }
+  };
+  xhr.send();
+}
+
+export function ajaxPost(url, type, data, success, error) {
+  var xhr = new XMLHttpRequest();
+  xhr.open('POST', url, true);
+  xhr.responseType = type;
+  xhr.setRequestHeader("Content-Type", "application/json");
+  xhr.setRequestHeader("Accept", "application/json");
+  xhr.setRequestHeader("X-Requested-With", "XMLHttpRequest");
+  xhr.setRequestHeader("X_CSRF_TOKEN", document.querySelector('meta[name=csrf-token]').getAttribute('content'));
+  xhr.onload = function() {
+    var status = xhr.status;
+    var response = xhr.response;
+  };
+  xhr.send(JSON.stringify(data));
+}
+
+export function updateUrl(propKey, propValue) {
```

Comment:
That is the direction this is taking. I'm giving these components a flag of persistent, where they can mangle the url. We'll see how it takes. Downside with a url store is that different components use different urls to access resources (a search is a different endpoint than saving a search for instance). Thoughts?